src/payment/strategies/adyenv2/adyenv2-payment-strategy.ts

```diff
@@ -0,0 +1,289 @@
+import { PaymentArgumentInvalidError } from '../../errors';
+import isVaultedInstrument from '../../is-vaulted-instrument';
+import Payment, { HostedInstrument } from '../../payment';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import {
+    AdyenCardState,
+    AdyenCheckout,
+    AdyenComponent,
+    AdyenConfiguration,
+    AdyenError,
+    ResultCode,
+    ThreeDS2ComponentType,
+    ThreeDS2OnComplete,
+    ThreeDS2Result,
+} from './adyenv2';
+import AdyenV2PaymentInitializeOptions from './adyenv2-initialize-options';
+import AdyenV2ScriptLoader from './adyenv2-script-loader';
+
+export default class AdyenV2PaymentStrategy implements PaymentStrategy {
+    private _adyenCheckout?: AdyenCheckout;
+    private _stateContainer?: string;
+    private _adyenv2?: AdyenV2PaymentInitializeOptions;
+    private _adyenComponent?: AdyenComponent;
+    private _adyenOldComponent?: AdyenComponent;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _adyenV2ScriptLoader: AdyenV2ScriptLoader,
+        private _formPoster: FormPoster
+
+    ) {}
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { adyenv2 } = options;
+
+        if (!adyenv2) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.adyenv2" argument is not provided.');
+        }
+
+        const paymentMethod = this._store.getState().paymentMethods.getPaymentMethod(options.methodId);
+
+        if (!paymentMethod) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        this._adyenv2 = adyenv2;
+
+        const configuration: AdyenConfiguration = {
+            environment:  paymentMethod.initializationData.environment,
+            locale: this._getLocale(),
+            originKey: paymentMethod.initializationData.originKey,
+            paymentMethodsResponse: paymentMethod.initializationData.paymentMethodsResponse,
+        };
+
+        return this._adyenV2ScriptLoader.load(configuration)
+            .then(adyenCheckout => {
+                this._adyenCheckout = adyenCheckout;
+
+                const adyenComponent = this._adyenCheckout.create(paymentMethod.method, {
+                        ...adyenv2.options,
+                        onChange: (state: AdyenCardState) => {
+                            this._updateStateContainer(state);
+                        },
+                    });
+
+                adyenComponent.mount(`#${adyenv2.containerId}`);
+                if (this._adyenComponent) {
+                    this._adyenOldComponent = this._adyenComponent;
```

Comment:
Ok, the reason that drove us to do this is because Adyen supports multiple APMs i.e. `scheme` & `bcmc`, and whem we switch from scheme to bcmc, initialize method (for bcmc) is called before deinitialize (for scheme) and that induce to unmount the existing component in the right moment (bcmc in this case which is already mounted) and the payment section keep as blank. Do you know how to keep this process async to support switch providers those use the same component and call first deinitialize (to unmount the first component) and after that initialize (to mount the last component) ??

Reply:
scheme & bcmc is using different containers, but the same instace as I already see in the method you linked, let me try doing that change and see if the flow works

Reply:
I finally did a success test with this line: 
`[paymentMethod.gateway, paymentMethod.id].filter(Boolean).join('-')`